themes/theme-gmd/pages/Product/components/ImageSlider/index.jsx

```diff
@@ -22,13 +22,32 @@ const resolutions = [
   },
 ];
 
+/**
+ * reformats the images array to group pictures by index, not by format
+ * @param {Array} images array of format images
+ * @returns {Array}
+ */
+const getImagesByIndex = (images) => {
+  const imagesByIndex = [];
+
+  images.forEach((format) => {
+    if (!format.sources || !format.sources.length) return;
+    format.sources.forEach((src, index) => {
+      if (!imagesByIndex[index]) imagesByIndex[index] = [];
+      imagesByIndex[index].push(src);
+    });
+  });
+
+  return imagesByIndex;
+};
+
 /**
  * The product image slider component.
  * @param {number} currentSlide The index of the current visible slide.
  */
 class ImageSlider extends PureComponent {
   static propTypes = {
-    images: PropTypes.arrayOf(PropTypes.string),
+    images: PropTypes.arrayOf(PropTypes.shape()),
```

Comment:
Since the determination of the `images` prop changed, and a new array is created on every selector call, the internal `shouldComponentUpdate` mechanism of the `PureComponent` doesn't work anymore. So you need to transform the component to a normal one with a custom `shouldComponentUpdate`